app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,30 @@
+require 'will_paginate/array'
+
+class RecipesController < ApplicationController
+
+  def home; end
+
+  def index
+    @query = params[:query]
+
+    if @query == "" || @query == nil
+      flash[:status] = :failure
+      flash[:message] = "Please enter the name of a recipe."
+      redirect_back(fallback_location: root_path)
+    else
+      recipes = EdamamWrapper.search_recipes(@query)
+      @recipes = recipes.paginate(page: params[:page], per_page: 12)
+    end
```

Comment:
You've written your API wrapper to raise an error if the API call fails, but you're not looking for an error here. You should wrap this call in a `begin`/`rescue`.